engine/access/ingestion/engine.go

```diff
@@ -337,7 +370,124 @@ func (e *Engine) processAvailableExecutionReceipts(ctx context.Context) error {
 		if err := e.handleExecutionReceipt(msg.OriginID, receipt); err != nil {
 			return err
 		}
+
+		// notify, to fetch and store transaction result error messages for block
+		e.txResultErrorMessagesChan <- receipt.BlockID
+	}
+}
+
+// processTransactionResultErrorMessages handles error messages related to transaction
+// results by reading from the error messages channel and processing them accordingly.
+//
+// This function listens for messages on the txResultErrorMessagesChan channel and
+// processes each transaction result error message as it arrives.
+//
+// No errors are expected during normal operation.
+func (e *Engine) processTransactionResultErrorMessages(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+	ready()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case blockID := <-e.txResultErrorMessagesChan:
```

Comment:
this should work for the happy case where the node is caught up with the network and receives at least one receipt for every block. It will miss data for cases where the node is offline or drops any messages. Currently, that would require the operator then take some manual action to backfill the data.

I think we need some additional tracking to ensure all blocks are handled. This wasn't included in the original design because it was handled already by the indexer engine.

How about adding another trigger point in the finalized block processing logic. When iterating the sealed blocks
https://github.com/onflow/flow-go/blob/3beefb84c15c3ebedb79154b8b63ad42f6102eb0/engine/access/ingestion/engine.go#L396-L401

Instead of pushing to the channel, synchronously call `handleTransactionResultErrorMessages`

Since there is already a jobqueue backing the finalized block processing, we're guaranteed that every block is processed, and this will add a follow up check in the event errors were missed for any blocks. It still won't guarantee that we get all error messages, but I think that's OK for now

Reply:
[3a9474b](https://github.com/onflow/flow-go/pull/6468/commits/3a9474b2a9005775f250f177473d0c9ac21223c6)